pkg/clusters/addons/kong/kong.go

```diff
@@ -146,11 +167,44 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		)
 	}
 
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
 	// do the deployment and install the chart
 	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
-	args = append(args, "--create-namespace", "--namespace", a.namespace)
-	args = append(args, a.deployArgs...)
+	args = append(args, "--namespace", a.namespace)
+	if a.enterprise {
+		if a.license != "" && a.license == KongEnterpriseTestLicense {
+			if err := deployKongEnterpriseLicenseSecret(ctx, cluster, a.namespace, KongEnterpriseTestLicense); err != nil {
+				return err
+			}
+		}
+
+		if err := prepareSecrets(ctx, a.namespace); err != nil {
+			return err
+		}
+
+		args = append(args, "--version", "2.3.0", "-f", "https://raw.githubusercontent.com/Kong/charts/main/charts/kong/example-values/minimal-k4k8s-with-kong-enterprise.yaml")
+		license := fmt.Sprintf("enterprise.license_secret=%s", a.license)
+		password := fmt.Sprintf("env.kong_password=%s", a.kongAdminPassword)
+		args = append(args, "--set", "admin.type=LoadBalancer", "--set", "admin.annotations.konghq.com/protocol=http", "--set", "enterprise.rbac.enabled=true",
+			"--set", "env.enforce_rbac=on", "--set", "ingressController.enabled=false",
```

Comment:
There is already a `WithProxyOnly` method on the builder which allows the caller to configure the Kong addon deployment with the controller disabled. We shouldn't implicitly do this as part of enterprise configuration.

```suggestion
			"--set", "env.enforce_rbac=on",
```

Reply:
04f3f25c7ee8e95e38ee39a9b1ce95298c65e388